.github/workflows/build_workflow.yml

```diff
@@ -1,18 +1,104 @@
 name: Build workflow
 run-name: Build workflow
-on: [push]
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - master
+  pull_request:
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
 jobs:
-  Explore-GitHub-Actions:
+  test:
     runs-on: ubuntu-latest
+    env:
+      TEST_REDIS: redis://redis:6379
+    container:
+      image: debian:bullseye
+    services:
+      redis:
+        image: redis
     steps:
-      - run: echo "🎉 The job was automatically triggered by a ${{ github.event_name }} event."
-      - run: echo "🐧 This job is now running on a ${{ runner.os }} server hosted by GitHub!"
-      - run: echo "🔎 The name of your branch is ${{ github.ref }} and your repository is ${{ github.repository }}."
-      - name: Check out repository code
-        uses: actions/checkout@v3
-      - run: echo "💡 The ${{ github.repository }} repository has been cloned to the runner."
-      - run: echo "🖥️ The workflow is now ready to test your code on the runner."
-      - name: List files in the repository
+      -
+        name: Install Essential Packages
+        run: | 
+          apt-get update && apt-get install -y --no-install-recommends libssl-dev cpanminus make gcc openssl \
+          zlib1g-dev pkg-config git libpq-dev apt-transport-https ca-certificates \
+          apt-utils wget curl gnupg2 lsb-release
+          git config --global --add safe.directory "$GITHUB_WORKSPACE"
+      -
+        name: Install Postgres
         run: |
-          ls ${{ github.workspace }}
-      - run: echo "🍏 This job's status is ${{ job.status }}."
+          wget --quiet -O - https://www.postgresql.org/media/keys/ACCC4CF8.asc | apt-key add -
+          echo "deb https://apt.postgresql.org/pub/repos/apt $(lsb_release -cs)-pgdg main" > /etc/apt/sources.list.d/pgdg.list
+          apt-get update
+          apt-get install -y postgresql-14 
+      -
+        name: Checkout Code
+        uses: actions/checkout@v4
+      -
+        name: Install Perl Dependencies
+        run: |
+          cpanm --quiet --notest Dist::Zilla Dist::Zilla::App::Command::cover Devel::Cover::Report::Coveralls
+          cpanm --quiet --notest $(dzil authordeps --missing)
+          cpanm -n --installdeps .
+          cpanm --quiet --notest $(dzil listdeps --author --missing)
+      -
+        name: Run Smoke Test
+        run: dzil smoke --release --author
+      -
+        name: Run Coverage Test
```

Comment:
I think you need to pass GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} as env to submit coverage report to coveralls. https://github.com/regentmarkets/github-actions/blob/master/.github/workflows/code_coverage.yml#L25C7-L25C48